src/message.rs

```diff
@@ -176,38 +176,39 @@ impl MimeMessage {
     /// Finally, update the `Content-Type` header with the `boundary`.
     /// It does not update the `message_type`
     pub fn update_headers(&mut self) {
-
+        // Since a single header can not be replace, clone all the headers
+        // except th `Content-Type`, which needs will be added later.
+        let mut headers = self.headers_without_content_type();
+        // Instead of returning the value of the `Content-Type` as a tuple,
+        // return it as String since it could have params.
+        let mut ct_value: String;
+        if let Some(message_type) = self.message_type {
+            // Use `message_type` as `Content-Type` if present
+            ct_value = message_type.to_string();
```

Comment:
@link2xt you're right, it's backward-compatible, we even never get here if `self.children.len() == 0`